src/XMakeBuildEngine/BackEnd/BuildManager/BuildParameters.cs

```diff
@@ -1007,56 +1003,16 @@ private void FindMSBuildExe()
             EnsureToolsets();
             Toolset currentToolset = _toolsetProvider.GetToolset(MSBuildConstants.CurrentToolsVersion);
 
-            if (currentToolset != null)
+            if (currentToolset != null && !string.IsNullOrEmpty(currentToolset.ToolsPath))
             {
-                path = Path.Combine(currentToolset.ToolsPath, "MSBuild.exe");
-                if (path != null && CheckMSBuildExeExistsAt(path))
+                location = Path.Combine(currentToolset.ToolsPath, "MSBuild.exe");
+                if (CheckMSBuildExeExistsAt(location))
                 {
-                    _nodeExeLocation = path;
-                    return;
+                    return location;
                 }
             }
 
-            // Search in the location of any assemblies we have loaded.
-            foreach (System.Reflection.Assembly assembly in AppDomain.CurrentDomain.GetAssemblies())
```

Comment:
This fallback seems like it might possibly be relevant. Have you shown that it's no longer needed somehow?

(I love the idea of getting rid of it! Just want to be cautious.)